src/internal/operations/backup_test.go

```diff
@@ -365,6 +369,109 @@ func TestBackupOpUnitSuite(t *testing.T) {
 	suite.Run(t, &BackupOpUnitSuite{Suite: tester.NewUnitSuite(t)})
 }
 
+func checkPopulatedInner(v reflect.Value) error {
+	if v.IsZero() {
+		return clues.New("zero-valued field")
+	}
+
+	if v.Kind() != reflect.Struct {
+		return nil
+	}
+
+	var errs *clues.Err
+
+	for i := 0; i < v.NumField(); i++ {
+		f := v.Field(i)
+
+		if err := checkPopulatedInner(f); err != nil {
+			errs = clues.Stack(errs, clues.Wrap(err, fmt.Sprintf("field at index %d", i)))
+		}
+	}
+
+	return errs.OrNil()
+}
+
+func checkPopulated(t *testing.T, input control.Options) {
+	err := checkPopulatedInner(reflect.ValueOf(input))
+	require.NoError(t, err, clues.ToCore(err))
+}
+
+func (suite *BackupOpUnitSuite) TestBackupOperation_OptionPassing() {
```

Comment:
The purpose of this test is a little obtuse by code alone.  I think if you added a comment here that matches, more or less, what you wrote for the PR description, it would help convey purpose and clarity.

Separately: nit on test func naming: `NewBackupOperation` is what's being tested (thus uppercase).  `configsMatchOperationOptions` is the test focus, but is not a func or struct (thus lowercase).
```suggestion
func (suite *BackupOpUnitSuite) TestNewBackupOperation_configsMatchOperationOptions() {
```